CI: Fix using `check_source` flags as bool

webknjaz opened the pull request:
Previously, those flags would sometimes end up having empty string values, which tends to break evaluating them as JSON. This patch adds `false` fallbacks to all such outputs.

This allows feeding them to `fromJSON()` without a fear of them causing surprising internal behaviors in the GitHub Actions CI/CD workflows platform itself [[1]]. The behavior observed was that some skipped jobs wouldn't show up in the workflow sidebar view at all, would display in the graph view as `Waiting for pending jobs` and in the `${{ needs }}` context, they would have a `result: failure` entry [[2]].

This should help make PRs like #121831 mergeable again.

[1]: https://github.com/python/cpython/pull/121766#issuecomment-2230023214
[2]: https://github.com/python/cpython/actions/runs/9950331379/job/27501637459?pr=121831#step:2:244

webknjaz committed:
🧪🚑 Fix using `check_source` flags as bool

Previously, those flags would sometimes end up having empty string
values, which tends to break evaluating them as JSON. This patch adds
`false` fallbacks to all such outputs.

This allows feeding them to `fromJSON()` without a fear of them
causing surprising internal behaviors in the GitHub Actions CI/CD
workflows platform itself [[1]]. The behavior observed was that
some skipped jobs wouldn't show up in the workflow sidebar view at
all, would display in the graph view as `Waiting for pending jobs`
and in the `${{ needs }}` context, they would have a
`result: failure` entry [[2]].

This should help make PRs like #121831 mergeable again.

[1]: https://github.com/python/cpython/pull/121766#issuecomment-2230023214
[2]: https://github.com/python/cpython/actions/runs/9950331379/job/27501637459?pr=121831#step:2:244

hugovk commented on .github/workflows/build.yml:
```diff
@@ -27,11 +27,33 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 10
     outputs:
+      # NOTE: Some of the referenced steps set outputs conditionally and there
```
Please could you delete all these NOTE prefixes? All comments are things to note :) 

```suggestion
      # Some of the referenced steps set outputs conditionally and there
```

Also, something I missed in https://github.com/python/cpython/pull/121766, please could you replace the ~forbidden "no-GIL" with "free-threaded"? We can do it in this PR.

webknjaz replied:
@hugovk could you point to where I kept the no-GIL reference? It's not visible in this diff. I remember having to use it to keep the commit messages under limits or smth like that, but I can't seem to find it anymore.

webknjaz replied:
Oh, it's in the inputs. Okay, will do.

webknjaz replied:
@hugovk done, and I'm marking this thread as resolved.

webknjaz committed:
🧪📝 Change reusable-windows input description

This updates it to avoid referencing to `no-GIL` in favor of the
official term `free-threading`.

hugovk commented on .github/workflows/build.yml:
```diff
@@ -27,11 +27,31 @@ jobs:
+      # may trigger all sorts of undefined and hard-to-debug behaviors in
+      # GitHub Actions CI/CD. To help with this, all of the outputs set here
+      # that are meant to be used as boolean flags (and not arbitrary strings),
+      # MUST have fallbacks with default values set. A common pattert would be
```
```suggestion
      # MUST have fallbacks with default values set. A common pattern would be
```

webknjaz replied:
:man_facepalming:

webknjaz replied:
Fixed. Marking as resolved.

hugovk reviewed (approved):
Thanks!
